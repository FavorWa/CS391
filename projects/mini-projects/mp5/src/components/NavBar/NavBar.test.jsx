import { test, expect } from 'vitest';
import renderer from 'react-test-renderer';
import NavBar from './NavBar.jsx';

test('The NavBar component should render', () => {
  const view = renderer
  .create(<NavBar/>)
  .toJSON();
  expect(view).toMatchSnapshot();
});

test('The NavBar component should render with a title', () => {
  const title = 'Test application';
  const component = renderer.create(<NavBar title={title} />);
  
  expect(component).toBeDefined();
});

test('The NavBar component should respond to button clicks', () => {
  const goBackHandler = () => {}; 
  const openFormHandler = () => {}; 
  const component = renderer.create(<NavBar goBack={goBackHandler} openForm={openFormHandler}/>);
  
  component.root.findByProps({children: '< Go Back'}).props.onClick();
  component.root.findByProps({children: '+ Add Review'}).props.onClick();
  expect(true).toBeTruthy();
  
});

