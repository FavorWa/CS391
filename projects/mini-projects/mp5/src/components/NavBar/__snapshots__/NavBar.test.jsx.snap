// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The NavBar component should render 1`] = `
<div
  className="sc-guDLey gtpvTe"
>
  <h2
    className="sc-dmyCSP jbzSYl"
  />
</div>
`;
